### ISSUE TITLE
Request Control Filters

### ISSUE THREAD

**Issue by crssi** [ghacksuserjs/ghacks-user.js#149]
This is [Request Control](https://addons.mozilla.org/firefox/addon/requestcontrol/) | [GitHub](https://github.com/tumpio/requestcontrol) 
This is our [Request Control Filters Wiki](https://github.com/ghacksuserjs/ghacks-user.js/wiki/3.5-Request-Control)

Post your Request Control Filter-fu here

Note: Examples below are for discussion / testing - use/test at your leisure, post feedback. As really cool filters emerge, we will put them in the wiki

**Comment by crssi** [comment]
Pure URL extension replacement:

**Pattern:** Any URL
**Types:** Document, Embedded document (not sure for second one)
**Action:** Filter
**Filter URL:** Off
**Trim URL Parameters:** ref_, utm_source, utm_medium, utm_term, utm_content, utm_campaign, utm_reader, utm_place, ga_source, ga_medium, ga_term, ga_content, ga_campaign, ga_place, yclid, _openstat, fb_action_ids, fb_action_types, fb_ref, fb_source, action_object_map, action_type_map, action_ref_map, ws_ab_test, btsid, algo_expid, algo_pvid, sid

**Comment by crssi** [comment]
Redirect to ref URL without tracking:

**Pattern:** Any URL
**Types:** Document, Embedded document (not sure for second one)
**Action:** Redirect
**Redirect To:** {href/(.?)ref=./$1}

Samples:
[https://www.amazon.com/AmazonBasics-Type-C-USB-Male-Cable/dp/B01GGKYQ02/ref=sr_1_1?s=amazonbasics&srs=10112675011&ie=UTF8&qid=1489067885&sr=8-1&keywords=usb-c](https://www.amazon.com/AmazonBasics-Type-C-USB-Male-Cable/dp/B01GGKYQ02/ref=sr_1_1?s=amazonbasics&srs=10112675011&ie=UTF8&qid=1489067885&sr=8-1&keywords=usb-c)

**Comment by crssi** [comment]
Remove the crap and possible tracking over URL manipulation after images:

**Pattern scheme:** http/https
**Pattern host:** *
**Pattern path:** .jpg?, .gif?, .png?, .svg?
**Types:** Image
**Action:** Filter
**Filter URL:** Off
**Trim URL Parameters:** Trim all

**Comment by crssi** [comment]
Facebook redirect without visiting facebook:

**Pattern scheme:** http/https
**Pattern host:** l.facebook.com
**Pattern path:** *u=*
**Types:** Document, Embedded document (not sure for second one)
**Action:** Filter
**Filter URL:** On

Sample:
[https://l.facebook.com/l.php?u=https%3A%2F%2Fwww.fsf.org%2Fcampaigns%2F&h=ATP1kf98S0FxqErjoW8VmdSllIp4veuH2_m1jl69sEEeLzUXbkNXrVnzRMp65r5vf21LJGTgJwR2b66m97zYJoXx951n-pr4ruS1osMvT2c9ITsplpPU37RlSqJsSgba&s=1](https://l.facebook.com/l.php?u=https%3A%2F%2Fwww.fsf.org%2Fcampaigns%2F&h=ATP1kf98S0FxqErjoW8VmdSllIp4veuH2_m1jl69sEEeLzUXbkNXrVnzRMp65r5vf21LJGTgJwR2b66m97zYJoXx951n-pr4ruS1osMvT2c9ITsplpPU37RlSqJsSgba&s=1)

**Comment by Thorin-Oakenpants** [comment]
https://thechive.files.wordpress.com/2017/06/dd0523324267789bb8beecc5d7914970.jpg?quality=85&strip=info&w=600

didn't work

**Comment by Thorin-Oakenpants** [comment]
I'll have to check these out and test them tomorrow. Shame there's no import feature (yet)

**Comment by crssi** [comment]
Funny... I have to look into it too.
Anyway, I have updated for CSS too... see change up.
One site is particulary full of this crap to test... see updated post.

**Comment by crssi** [comment]
Hah... resolved, done.
Have phun

**Comment by crssi** [comment]
The author is working now on import/export. ;)

**Comment by Atavic** [comment]
[Here](https://github.com/newhouse/url-tracking-stripper) some Google's Urchin Tracking Modules that aren't listed in **[Trim URL Parameters list](https://github.com/ghacksuserjs/ghacks-user.js/issues/149#issuecomment-310173763)**: 

    utm_name
    utm_cid
    utm_reader
    utm_viz_id
    utm_pubreferrer
    utm_swu

Also, url-tracking-stripper has a few more: 

    ICID
    icid
    _hsenc
    _hsmi
    mkt_tok
    sr_share
    vero_conv
    vero_id
    nr_email_referer
    ncid

**Comment by earthlng** [comment]
Thanks @crssi !! :+1: 

a couple nits if you don't mind...

* the ga_ stuff probably needs to include XHR
* `{href/(.?)ref=./$1}` doesn't seem right. I'm pretty sure this always results in a single `h`, right? `(.?)`
and wouldn't it break something like http://foo.bar/redirect?**href**=bar.foo/index.html
to give you an example, this is what I came up with for a bing-redirect:
`{search/^\?.*&?q=(.+?)(&.*|#.*|$)/https://www.ixquick.eu/do/dsearch?query=$1&cat=web&pl=opensearch&language=english}` - it covers everything I could think of, namely:
- http://www.bing.com/search**?q=crssi**&go=Submit+Query&qs=ds&form=QBLH
- http://www.bing.com/search?go=Submit+Query**&q=crssi**&qs=ds&form=QBLH
- http://www.bing.com/search?go=Submit+Query&qs=ds&form=QBLH**&q=crssi**
- http://www.bing.com/search?q=crssi
- http://www.bing.com/search?go=Submit+Query&qs=ds&form=QBLH**&q=crssi#anchor1
and even then, I'm sure I missed *something* - regexes are almost impossible to get right

2 more from my small collection:
- `Any URL` `CSP report` `Block`

Pattern scheme: http/https
Pattern host: youtu.be
Pattern path: *
Types: Document, (Embedded document)
Action: Redirect
Redirect To: `https://www.youtube.com/watch?v={pathname:1}`

**Comment by Thorin-Oakenpants** [comment]
@earthlng Have you installed and looked at Redirector yet? It comes with no preset "redirects" but allows exporting/importing as txt (or other files) which would make it so much easier to post items. Does Request Control offer anything that Redirector doesn't or vice versa?

**Comment by crssi** [comment]
You are right. Need to rethink the REF one, it was fishy to me also and I don't like it at all.... will put a "Under construction" note on that post.

The ga_ stuff, should we separate into another filter extended with XHR or just enable the XHR on the current one. I will enable here now and see if I will get any breakage... will also put a note in that post.

I have another "nasty" one, which could replace "Skip Redirect" extension, but it is also fishy... for now I am using Redirector, since I can use regex (not that I like it, but can't see better way.

I can place the nasty one in another post together with the Redirector approach for your critics?

**Comment by Thorin-Oakenpants** [comment]
Umm .. these are all under construction until they land in the wiki :)

**Comment by earthlng** [comment]
> Have you installed and looked at Redirector yet?

nope

> Does Request Control offer anything that Redirector doesn't or vice versa?

IDK, you tell me. I'm pretty happy with Request Control and need a lot more convincing to try Redirector ;)
Judging from the name alone, the former seems better suited for what I want, rather than just "redirects"

**Comment by Thorin-Oakenpants** [comment]
I'm asking your expert opinion dude, have a look at it... I would like to know if it offers anything Request Control doesn't. These 2 WEs seem very similar (redirector seems like a misnomer, as it can manipulate the strings just like request control, yes regex allowed), and hence we should work out which one we will recommend (and add a notation to the other <sup>1</sup> )

I see no point in maintaining two wiki's for rules, or running two extensions which make cause issues. Just take a look at it and give me your opinion

**Comment by earthlng** [comment]
I already edited my post dude - chillax mate :)
I'll probably end up using both

**Comment by crssi** [comment]
Redirector has some problems, when clicking URL from external program it doesn't always work as intended.
Another thingy is that Redirector parses input only by one filter (just my observation) and doesn't pass the result over filters again unitl nothing left to filter (it happens that you want to filter out more than only one thing).
Otherwise it has also exclusions which is great and better string manipulation over regex.
Both are very good.

**Comment by crssi** [comment]
@Thorin-Oakenpants do you have any sample for ga_ stuff, so I can dig in?
After tomorrow I will be offline for about 5 days.

**Comment by Thorin-Oakenpants** [comment]
@crssi I too will be rather busy the next few days (gotta pay for that wall somehow)... the thing is there is sooooo much of this click-tracking BS going on with ~~so many~~ almost every site~~s~~, but its also so easy to break things. I'm not a code rilla :monkey:  like earthlng, so all of this flies over my head. I just want rules I can stick in (preferably a storage.js file with hundreds of rules, or an import file). Things like that imdb example is just that, an example (and its internal imdb tracking), but we could do with stopping outgoing tracks from AMO, reddit etc, but then we would also end up with thousands of rules. We need some really generic ones, but I have no idea how bad the breakage could get.

No idea on how to find and test all the various strings - surely if it works for one, it must work for the other(s) ?

**Comment by crssi** [comment]
[Updated RedirectTo](https://github.com/ghacksuserjs/ghacks-user.js/issues/149#issuecomment-310175387)

does not cover @Thorin-Oakenpants  sample: http://foo.bar/redirect?href=bar.foo/index.html

**Comment by crssi** [comment]
Now a nasty one... trying to mimic part of "Skip redirect"
NOTE: this one can and will make some login breakage... for tests and brainstorming  only

It consists 2 filter rules.
First one does the skipping and the second one whitelists login pages to avoid breakage.

**Pattern:** Any URL
**Types:** Document, Embedded document
**Action:** Filter
**Filter URL:** On

**Pattern scheme:** http/https
**Pattern host:** *
**Pattern path:** \*signin\*, \*signout\*, \*login\*, \*logout\*, \*logon\*, \*logoff\*, \*auth\*, \*account\*, \*eBayISAPI.dll\* (maybe you can add also \*option\*, but I don't like it)
**Types:** Document, Embedded document
**Action:** Whitelist

Samples:
**Working redirect**
[https://outgoing.prod.mozaws.net/v1/b928e4237edbbdd2646a3971d2e6b514aee033c10f3f4c49415bf93096405f38/http%3A//www.google.com/chrome/%3Fi-would-rather-use-firefox=http%253A%252F%252Fwww.mozilla.org/](https://outgoing.prod.mozaws.net/v1/b928e4237edbbdd2646a3971d2e6b514aee033c10f3f4c49415bf93096405f38/http%3A//www.google.com/chrome/%3Fi-would-rather-use-firefox=http%253A%252F%252Fwww.mozilla.org/)

[www.google.com/chrome/?i-would-rather-use-firefox=http%3A%2F%2Fwww.mozilla.org/](www.google.com/chrome/?i-would-rather-use-firefox=http%3A%2F%2Fwww.mozilla.org/)

**Working login pages**
[https://signin.ebay.de/ws/eBayISAPI.dll?SignIn&UsingSSL=1&pUserId=&co_partnerId=2&siteid=77&ru=http%3A%2F%2Fmy.ebay.de%2Fws%2FeBayISAPI.dll%3FMyEbayBeta%26MyEbay%3D%26gbh%3D1%26guest%3D1&pageType=3984](https://signin.ebay.de/ws/eBayISAPI.dll?SignIn&UsingSSL=1&pUserId=&co_partnerId=2&siteid=77&ru=http%3A%2F%2Fmy.ebay.de%2Fws%2FeBayISAPI.dll%3FMyEbayBeta%26MyEbay%3D%26gbh%3D1%26guest%3D1&pageType=3984)

[https://www.amazon.com/ap/signin?_encoding=UTF8&openid.assoc_handle=usflex&openid.claimed_id=http%3A%2F%2Fspecs.openid.net%2Fauth%2F2.0%2Fidentifier_select&openid.identity=http%3A%2F%2Fspecs.openid.net%2Fauth%2F2.0%2Fidentifier_select&openid.mode=checkid_setup&openid.ns=http%3A%2F%2Fspecs.openid.net%2Fauth%2F2.0&openid.ns.pape=http%3A%2F%2Fspecs.openid.net%2Fextensions%2Fpape%2F1.0&openid.pape.max_auth_age=0&openid.return_to=https%3A%2F%2Fwww.amazon.com%2Fgp%2Fyourstore%2Fcard%3Fie%3DUTF8%26ref_%3Dcust_rec_intestitial_signin](https://www.amazon.com/ap/signin?_encoding=UTF8&openid.assoc_handle=usflex&openid.claimed_id=http%3A%2F%2Fspecs.openid.net%2Fauth%2F2.0%2Fidentifier_select&openid.identity=http%3A%2F%2Fspecs.openid.net%2Fauth%2F2.0%2Fidentifier_select&openid.mode=checkid_setup&openid.ns=http%3A%2F%2Fspecs.openid.net%2Fauth%2F2.0&openid.ns.pape=http%3A%2F%2Fspecs.openid.net%2Fextensions%2Fpape%2F1.0&openid.pape.max_auth_age=0&openid.return_to=https%3A%2F%2Fwww.amazon.com%2Fgp%2Fyourstore%2Fcard%3Fie%3DUTF8%26ref_%3Dcust_rec_intestitial_signin)

[https://manage.autodesk.com/service/authentication/login?returnUrl=https%3A%2F%2Fmanage.autodesk.com%2Fcep%2F](https://manage.autodesk.com/service/authentication/login?returnUrl=https%3A%2F%2Fmanage.autodesk.com%2Fcep%2F)

[https://account.xiaomi.com/pass/serviceLogin?callback=http%3A%2F%2Fglobal.mi.com%2Fen%2Flogin%2Fcallback%3Ffollowup%3Dhttp%253A%252F%252Fwww.mi.com%252Fen%252F%26sign%3DZDk4YmY0MTRkMThmODExYTE0MDljYWRmYzczZmNjOGZjNDAzNzU0Mg%2C%2C&sid=mi_overseaen&_locale=en](https://account.xiaomi.com/pass/serviceLogin?callback=http%3A%2F%2Fglobal.mi.com%2Fen%2Flogin%2Fcallback%3Ffollowup%3Dhttp%253A%252F%252Fwww.mi.com%252Fen%252F%26sign%3DZDk4YmY0MTRkMThmODExYTE0MDljYWRmYzczZmNjOGZjNDAzNzU0Mg%2C%2C&sid=mi_overseaen&_locale=en)

[https://login.live.com/login.srf?wa=wsignin1.0&rpsnv=13&ct=1495461484&rver=6.7.6643.0&wp=MBI_SSL_SHARED&wreply=https:%2F%2Fmail.live.com%2Fdefault.aspx%26lc%3D1033%26id%3D64855%26mkt%3Den-us%26cbcxt%3Dmai%26lc%3D1033%26id%3D64855%26mkt%3Den-us%26cbcxt%3Dmai%26lc%3D1033%26id%3D64855%26mkt%3Den-us%26cbcxt%3Dmai%26lc%3D1033%26id%3D64855%26mkt%3Den-us%26cbcxt%3Dmai&lc=1033&id=64855&mkt=en-us&cbcxt=mai](https://login.live.com/login.srf?wa=wsignin1.0&rpsnv=13&ct=1495461484&rver=6.7.6643.0&wp=MBI_SSL_SHARED&wreply=https:%2F%2Fmail.live.com%2Fdefault.aspx%26lc%3D1033%26id%3D64855%26mkt%3Den-us%26cbcxt%3Dmai%26lc%3D1033%26id%3D64855%26mkt%3Den-us%26cbcxt%3Dmai%26lc%3D1033%26id%3D64855%26mkt%3Den-us%26cbcxt%3Dmai%26lc%3D1033%26id%3D64855%26mkt%3Den-us%26cbcxt%3Dmai&lc=1033&id=64855&mkt=en-us&cbcxt=mai)

**Currently broken login pages**
[https://login.leagueoflegends.com/?region=eune&lang=en_PL&redirect_uri=http%3A%2F%2Feune.leagueoflegends.com%2F](https://login.leagueoflegends.com/?region=eune&lang=en_PL&redirect_uri=http%3A%2F%2Feune.leagueoflegends.com%2F)

**Comment by crssi** [comment]
Crap... at least "path" definition is case sensitive. I have placed an issue about that.

**Comment by Atavic** [comment]
:-1: [Redirector](https://addons.mozilla.org/en-US/firefox/addon/redirector/reviews/883079/)

**Comment by crssi** [comment]
^^ Ahh, this now perfectly explains my observation about Redirector and what I didn't like about it.
But how then uBo, uMatrix and Request Controls are dealing perfectly in those cases.

**Comment by Atavic** [comment]
>URL path definition is case sensitive

The absolute URL path: `http://example.com/data/` is not case sensitive, as per [RFC 3986](https://tools.ietf.org/html/rfc3986#page-40).

While all that comes after could be case sensitive, depending on web server settings.

Everything after the RFC Standard is an [unclear mess](https://github.com/nodejs/node-v0.x-archive/issues/6000).

**Comment by earthlng** [comment]
@crssi wrote:

> do you have any sample for ga_ stuff

any site that uses GoogleAnalytics, fe ghacks. GA should already be blocked by uBO or uMatrix or whatnot and I'm not sure we really need a filter for that stuff.

earthlng wrote

> My way too complex bing regex is probably not necessary. It looks like the addon strips additional parameters automatically

Yeah, it doesn't, at least not for "Action: Redirect". Redirector doesn't make it easier either.
But I realized I don't need to account for anchors because they are not part of the "search" parameter, which results in this slightly adjusted bing-redirect: 

old: `{search/^\?.*&?q=(.+?)(&.*|#.*|$)/https://www.ixquick.com/do/dsearch?query=$1&cat=web&pl=opensearch&language=english}`
new: `{search/^\?.*&?q=(.+?)(&.*|$)/https://www.ixquick.com/do/dsearch?query=$1&cat=web&pl=opensearch&language=english}`
or: `https://www.ixquick.com/do/dsearch?query={search/^\?.*&?q=(.+?)(&.*|$)/}&cat=web&pl=opensearch&language=english`
... whichever format you prefer.

**Comment by earthlng** [comment]
> We need some really generic ones, but I have no idea how bad the breakage could get.

I would stay away from generic ones if you care about efficiency and no breakage.

> Does Request Control offer anything that Redirector doesn't or vice versa?

Redirector

* pros
  * example result
  * control over processing -> "Process matches"
  * simpler terms (iframe instead of embedded document)
  * import/export rules
  * Exclude Pattern (you can probably achieve the same in RC with a bit of regex-fu)
  * (custom description)
* cons
  * simpler terms ("other" instead of CSP report (which I guess is part of "other"), etc)
  * no direct access to only parts of the url (ie RC's "parameters")
  * doesn't support multiple domains/paths in one rule
  * no "TLD's" support (fe. see default google rule in RC)
  * no "block" request

Redirector's author said the addon is in "maintenance mode" and he won't add new features.
RC's author still seems to be very actively maintaining the addon and is open to add new features.

RC does some "magic" in the background which is great (and seems to work) but not very transparent.
fe. it strips additional parameters (&xyz=foo&bar=etc) with "Filter" rules and does urlencoding if necessary. I'd personally prefer if there was a bit more control over this, especially the encoding stuff.
Maybe we can lobby him/her to add that (with a default "auto" option to keep the current behavior)

**Comment by GitCurious** [comment]
1. https://github.com/ghacksuserjs/ghacks-user.js/issues/149#issuecomment-310175387

This appears to be breaking some internal Amazon links (such as "Warehouse Deals" for example),
presumably it`s the ref= part.

2. https://github.com/ghacksuserjs/ghacks-user.js/issues/149#issuecomment-310431067

With regard to this, how exactly are these "2" filter rules applied ?
Separately ?, must they be in a specific order ?

This stuff gives me a headache at the best of times - intriguing stuff though !

**Comment by Thorin-Oakenpants** [comment]
I have no idea - it all does my head in. But as earthlng mentioned, generic catch-all ones will probably cause breakage eg the removal of ref will break searching on imdb `http://www.imdb.com/find?ref_=nv_sr_fn&q=firefox&s=all` as you can see when I search for the film Firefox. Bound to be hundreds of thousands / millions of these sort of things/exceptions

Some discussion going on here: https://www.reddit.com/r/firefox/comments/6j6w0v/how_to_automatically_clean_up_urls_by_removing/  (I see you @Forsaked .. we're always watching :) )

**Comment by Forsaked** [comment]
OT: @Thorin-Oakenpants Good to know, looks like im not John Cena!

**Comment by earthlng** [comment]
FYI atm there's a bug that a redirected Document still gets added to the history. However the request doesn't touch the network. (that was fixed in 54 and also backported to ESR [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1345893))
re: https://www.reddit.com/r/firefox/comments/6j6w0v/how_to_automatically_clean_up_urls_by_removing/djc7w8v/

**Comment by wfRbeHBASpPKniSGBUvZ** [comment]
The *.jpg?*, *.png?* ... image filters break images on www.kickstarter.com.

**Comment by Atavic** [comment]
Try whitelisting `https://ksr-ugc.imgix.net/`

**Comment by crssi** [comment]
Will keep eye on this addon, since it has a lot of potential, but will hold until case insesitive isn't sorted out.

**Comment by crssi** [comment]
@GitCurious REF is tricky and WILL make breakage.
There are no orders in RC.
The first rule takes the last URL in the URL (damn if this makes sense :))
And the second skips this action when you hit some login pages.

**Comment by crssi** [comment]
There are also some other issues. I have placed an Issue on RC github.

**Comment by Thorin-Oakenpants** [comment]
You can now import/export rules to a json file. It exports the lot, but each rule is nestled between `{ }`.

We should use this format from now on (and I will add instructions at the top of the wiki page - i.e save the code snippet into a whatever.json and import). My only issue if that with loads of these, the page will get uber long

example rule for imdb
```
[
  {
    "pattern": {
      "scheme": "*",
      "host": [
        "www.imdb.com"
      ],
      "path": [
        "title/*",
        "name/*",
        "character/*"
      ]
    },
    "types": [
      "main_frame"
    ],
    "action": "filter",
    "active": true,
    "paramsFilter": {
      "values": [
        "ref_"
      ],
      "pattern": "ref_"
    },
    "skipRedirectionFilter": true
  }
]
```

**Comment by Thorin-Oakenpants** [comment]
^^ I wonder if we can compress it up to less lines (I don't why that wouldn't work) and it still imports nicely

**Comment by crssi** [comment]
Sure you can. Its just a JSON.
This is the same:
`[{"pattern":{"scheme":"*","host":["www.imdb.com"],"path":["title/*","name/*","character/*"]},"types":["main_frame"],"action":"filter","active":true,"paramsFilter":{"values":["ref_"],"pattern":"ref_"},"skipRedirectionFilter":true}]`

**Comment by Thorin-Oakenpants** [comment]
Yeah, I already tested it and modified the wiki page - go have a look

**Comment by crssi** [comment]
Damn cool... how do you do that?

**Comment by Thorin-Oakenpants** [comment]
^^ huh? I didn't do anything. I was using ``` and the string has no line breaks in it. You used a single ` and  this forces a word-wrap. I prefer yours as this means the code can be seen at a glance, and it's easier to select for copypasta

**Comment by crssi** [comment]
Ah, ok, I see now... thx :)

**Comment by Atavic** [comment]
@crssi Google has `&biw=` and others, see [RemoveGoogleTracking](https://github.com/sceneq/RemoveGoogleTracking).

**Comment by crssi** [comment]
Thank you @Atavic. Will take a look into in the next days

**Comment by crssi** [comment]
For redirect skipping RC is no match to [Skip Redirect (WE)](https://addons.mozilla.org/en-US/firefox/addon/skip-redirect/), especially now where author implemented same domain detection... see [https://github.com/sblask/webextension-skip-redirect/issues/30](https://github.com/sblask/webextension-skip-redirect/issues/30)

I will not try to mimic the same functionality on RC anymore.

For tracking removal basic the [Link Cleaner](https://addons.mozilla.org/en-US/firefox/addon/link-cleaner/) does a nice job, but RC can be used for more advanced rules or for tracking options that are not covered in LC.

**Comment by Atavic** [comment]
> 'biw', // offsetWidth
> 'bih', // offsetHeight

Apparently related to screen fingerprint.

**Comment by crssi** [comment]
@Atavic 
To to be sure... should I add to [https://github.com/ghacksuserjs/ghacks-user.js/issues/149#issuecomment-310173763](https://github.com/ghacksuserjs/ghacks-user.js/issues/149#issuecomment-310173763) the following:
biw, bih, ei, sa, ved, source, prmd, bvm, bav, psi, stick, dq, ech, gs_gbg, gs_rn, cp, scroll, vet, yv, ijn, iact, forward, ndsp, csi, tbnid, pbx, dpr, pf, gs_rn, gs_mss, pq, cp, oq, sclient, gs_l, aqs, psi

Observation: From what I see on the source code of [RemoveGoogleTracking](https://github.com/sceneq/RemoveGoogleTracking), its more complex than just to remove those parameters.

**Comment by crssi** [comment]
I have redesigned some rules and dropped some from the scratch and have issued few issues to @tumpio... specially [this one](https://github.com/tumpio/requestcontrol/issues/40).
When some more is known about the issue I will close this topic and open a new topic (referenced to this one) and post from scratch.

**Comment by crssi** [comment]
@Thorin-Oakenpants 
If you agree I would like to close this topic, since @tumpio is not active almost 2 months and there is better approach to deal with skipping redirect and url cleaning.
I can open new "post" with all together solution with setup I am using for skip redirect and url cleaning in the next few days?

**Comment by Thorin-Oakenpants** [comment]
Just leave this one open for now. `Request Control` & `Redirector` samples on the wiki are non-existent, and I would like to start harnessing their power. We definitely at some stage need a new shiny topic without all the clutter - eg broken rules/filters, duplicity over utms etc. These two extensions can do way more than `Skip Redirect` + `Neat URL / Link Cleaner` and I'd like some decent examples for their respective rules and filters wiki pages

**Comment by Thorin-Oakenpants** [comment]
Closing this. If anyone with the knowledge want to post a new topic with some DECENT examples for the wiki, then do so - key word EXAMPLE (this way others can use the example to work out how to create others for specific sites or whatever) - ps, no need to replicate things covered by Neat URL or Skip Redirect

**Comment by earthlng** [comment]
example to skip youtu.be links:
pattern: scheme: `http/https` host: `youtu.be` path: `*`
types: `Document`
action: `Redirect`
redirect to: `https://www.youtube.com/watch?v={pathname:1}`

example url: https://youtu.be/nqbUkThGlCo

Request Control supports direct access to [named parameters](https://github.com/tumpio/requestcontrol/blob/master/_locales/en/manual.md#list-of-named-parameters)
The named parameter "pathname" in the example link is `/nqbUkThGlCo` and `{pathname:1}` strips the slash using [substring extraction](https://github.com/tumpio/requestcontrol/blob/master/_locales/en/manual.md#substring-extraction)

**Comment by securingmom** [comment]
My love to all of you

**Comment by seascape** [comment]
Can anyone kindly help me with a rule that converts percent-encoded URLs to more human-readable (and working) URLs? An add-on mishap left me with a Firefox full of hundreds of percent-encoded tabs that I'd rather not unfurl manually and just trimming away the add-on's prefix and feeding Firefox the percent-encoded URL is not working (it instead does a Google search for the mess of percent-encoded text). 

I've poked at Request Control to convert these URLs back to normal but am out of my depth.

example of a tab URL:
`moz-extension://46e64bb4-2015-4b38-a64f-f732b4466147/replaced/index.html?state=redirect&title=Tehran%2C+Dec+29th+%3A+pics&url=https%3A%2F%2Fold.reddit.com%2Fr%2Fpics%2Fcomments%2F1pyrihs%2Ftehran_dec_29th%2F`

cleaned up output would be:
`https://old.reddit.com/r/pics/comments/1pyrihs/tehran_dec_29th/`

I need to discard everything from `moz-extension://` to `&url=` and then convert the rest from URL percent encoding. I think Request Control can do this but I'm unsure how to get the rule right. Appreciate any help.
